### ISSUE TITLE
Nystrom kernel approximation

### ISSUE THREAD

**Issue by st--** [trthatcher/MLKernels.jl#24]
I believe the Nystrom kernel approximation should satisfy the following invariant:
`nystrom(kernel, X, [1:size(X,1)]) == kernelmatrix(kernel, X)` (up to floating point inaccuracies)
The current implementation is definitely broken.
If I use just Base routines, as follows, I get what I believe is the correct approximation:

```
function basenystrom(kernel::Kernel, X::Matrix, xs::Vector)
    C = kernelmatrix(kernel, X, X[xs,:])
    D = C[xs,:]
    SVD = svdfact(D)
    DVC = diagm(1./sqrt(SVD[:S])) * SVD[:Vt] * C'
    MLKernels.syml(BLAS.syrk('U', 'T', 1, DVC))
end
```

But I don't understand your optimised version well enough to figure out where it's going wrong...

I've added this as test/test_approx.jl; have a look and have fun fiddling with your implementation until that test passes. ;-) It's not yet added to runtests.jl so it doesn't break the overall test.

**Comment by trthatcher** [comment]
Haha, thanks. That looks like the intended Nystrom method. I'll look into this

**Comment by st--** [comment]
Fixed in 1e3cc7ebd682ce6eaf000d59dacf96d9a3b1f147
